### PR TITLE
Add flake8, black, isort; GitHub action

### PR DESCRIPTION
**Pull request by eloquence**
- Adds flake8 linter and black/isort code formatters, and applies them to the existing code.
- Adds Makefile convenience targets.
- Adds GitHub actions to run all checks.

### CODE REVIEW

File: trackthenews/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-from .core import *
+from .core import main  # noqa: F401
```
Comment (eloquence) [inline trackthenews/__init__.py:1]: There's no need for a wildcard import as far as I can tell; `setup.py` only specifies `main` as an entry point.

File: trackthenews/core.py [hunk 2]
```diff
@@ -487,15 +469,17 @@ def initial_setup():
     else:
         try:
             os.makedirs(home)
-        except:
+        except Exception:
```
Comment (eloquence) [inline trackthenews/core.py:472]: This satisfies flake8's very reasonable check to not use bare exceptions (which get in the way of program interrupts and such).

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies
+        run: |
+          apt-get update && apt-get install --yes --no-install-recommends make python3-pip
+          pip install black==23.7.0 flake8==6.0.0 isort==5.12.0
```
Comment (eloquence) [inline .github/workflows/ci.yml:12]: Once we switch to `poetry`, we can create a dev dependency group for these - one of the benefits of `poetry`

File: trackthenews/core.py [hunk 4]
```diff
@@ -425,24 +413,20 @@ def setup_matchlist():
         with open(path, "w") as f:
             f.write("")
         print(
-            "A new matchlist has been generated at {path}. You can add case insensitive entries to match, one per line.".format(
-                **locals()
-            )
+            "A new matchlist has been generated at {path}."
+            " You can add case insensitive entries to match, one per line.".format(**locals())
```
Comment (harrislapiroff) [inline trackthenews/core.py:417]: Wow I did not realize this sort of multi-line formatting would work.

Future: This change didn't introduce this line, but probably we should eventually use an f-string here and in the other locations that use `**locals()` for string formatting data

**Review by harrislapiroff — APPROVED** [review]
LGTM and I didn't detect any regressions. Did notice some spots I think we should cleanup the code in the future in the process though.

**Pull request merged**